### PR TITLE
fix: CDK v2 integration tests should not use v1 feature flags

### LINKED ISSUE #1343: [bug] `integ:*:deploy` doesn't work with CDK v2
When using `awscdk.AwsCdkConstructLibrary` and AWS CDK v2 (the new default), `integ:*:deploy` scripts give me this error:

```
josh@josh-XPS-15-9550:~/proj/test-a$ yarn integ:foobar:deploy
yarn run v1.22.17
$ npx projen integ:foobar:deploy
👾 integ:foobar:deploy | rm -fr test/.tmp/foobar.integ/deploy.cdk.out
👾 integ:foobar:deploy | cdk deploy --app "ts-node -P tsconfig.dev.json test/foobar.integ.ts" --no-version-reporting --context aws-cdk:enableDiffNoFail=true --context @aws-cdk/aws-apigateway:usagePlanKeyOrderInsensitiveId=true --context @aws-cdk/core:enableStackNameDuplicates=true --context @aws-cdk/core:stackRelativeExports=true --context @aws-cdk/aws-ecr-assets:dockerIgnoreSupport=true --context @aws-cdk/aws-secretsmanager:parseOwnedSecretName=true --context @aws-cdk/aws-kms:defaultKeyPolicies=true --context @aws-cdk/aws-s3:grantWriteWithoutAcl=true --context @aws-cdk/aws-ecs-patterns:removeDefaultDesiredCount=true --context @aws-cdk/aws-rds:lowercaseDbIdentifier=true --context @aws-cdk/aws-efs:defaultEncryptionAtRest=true --context @aws-cdk/aws-lambda:recognizeVersionProps=true --context @aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021=true --context @aws-cdk/core:newStyleStackSynthesis=true --require-approval=never -o test/.tmp/foobar.integ/deploy.cdk.out
/home/josh/proj/test-a/node_modules/aws-cdk-lib/core/lib/feature-flags.ts:18
        throw new Error(`Unsupported feature flag '${featureFlag}'. This flag existed on CDKv1 but has been removed in CDKv2.`
              ^
Error: Unsupported feature flag '@aws-cdk/core:enableStackNameDuplicates'. This flag existed on CDKv1 but has been removed in CDKv2. CDK will now behave as the same as when the flag is enabled.
    at FeatureFlags.isEnabled (/home/josh/proj/test-a/node_modules/aws-cdk-lib/core/lib/feature-flags.ts:18:15)
    at new Stack (/home/josh/proj/test-a/node_modules/aws-cdk-lib/core/lib/stack.ts:207:47)
    at Object.<anonymous> (/home/josh/proj/test-a/test/foobar.integ.ts:4:15)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Module.m._compile (/home/josh/proj/test-a/node_modules/ts-node/src/index.ts:1371:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Object.require.extensions.<computed> [as .ts] (/home/josh/proj/test-a/node_modules/ts-node/src/index.ts:1374:12)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12)
Subprocess exited with error 1
👾 Task "integ:foobar:deploy" failed when executing "cdk deploy --app "ts-node -P tsconfig.dev.json test/foobar.integ.ts" --no-version-reporting --context aws-cdk:enableDiffNoFail=true --context @aws-cdk/aws-apigateway:usagePlanKeyOrderInsensitiveId=true --context @aws-cdk/core:enableStackNameDuplicates=true --context @aws-cdk/core:stackRelativeExports=true --context @aws-cdk/aws-ecr-assets:dockerIgnoreSupport=true --context @aws-cdk/aws-secretsmanager:parseOwnedSecretName=true --context @aws-cdk/aws-kms:defaultKeyPolicies=true --context @aws-cdk/aws-s3:grantWriteWithoutAcl=true --context @aws-cdk/aws-ecs-patterns:removeDefaultDesiredCount=true --context @aws-cdk/aws-rds:lowercaseDbIdentifier=true --context @aws-cdk/aws-efs:defaultEncryptionAtRest=true --context @aws-cdk/aws-lambda:recognizeVersionProps=true --context @aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021=true --context @aws-cdk/core:newStyleStackSynthesis=true --require-approval=never -o test/.tmp/foobar.integ/deploy.cdk.out" (cwd: /home/josh/proj/test-a)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I get a similar error when I try to run `integ:foobar:snapshot`.

These feature flags probably shouldn't be included when synthesizing AWS CDK v2 app.

Here's an SSCCE: https://github.com/misterjoshua/projen-integ-deploy-bug-cdkv2

To reproduce in the SSCCE run these commands:
```sh
yarn
yarn integ:foobar:deploy
````

### PR DESCRIPTION
**Pull request by misterjoshua**
Changes CDK v2 integration tests so that they don't use the v1 feature flags.

Fixes #1343

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/awscdk/integration-test.test.ts [hunk 1]
```diff
@@ -72,10 +73,38 @@ test('installs ts-node if needed', () => {
...
+  expect(tasks['integ:foo:deploy'].steps).toEqual(
+    expect.arrayContaining([
+      { exec: 'cdk deploy --app "ts-node -P tsconfig.dev.json test/foo.integ.ts" --no-version-reporting --require-approval=never -o test/.tmp/foo.integ/deploy.cdk.out' },
+    ]),
+  );
+  expect(tasks['integ:foo:snapshot'].steps).toEqual(
+    expect.arrayContaining([
+      { exec: 'cdk synth --app "ts-node -P tsconfig.dev.json test/foo.integ.ts" --no-version-reporting -o test/foo.integ.snapshot > /dev/null' },
+    ]),
+  );
```
Comment (Chriscbr) [inline test/awscdk/integration-test.test.ts:109]: I would prefer if we either made a more specific assertion (e.g. that the command does or doesn't pass a certain flag related to v2), or just replaced these with snapshots, like:

```ts
expect(output['.projen/tasks.json'].tasks['integ:foo:deploy']).toMatchSnapshot();
expect(output['.projen/tasks.json'].tasks['integ:foo:snapshot']).toMatchSnapshot();
```

**Review by Chriscbr — COMMENTED** [review]
@misterjoshua Looks good to me! Just one small comment on the unit test

File: test/awscdk/integration-test.test.ts (hunk 1 above)
Comment (misterjoshua) [reply test/awscdk/integration-test.test.ts:109]: Ok. I'll see what I can do here.

**Comment by misterjoshua** [comment]
@Chriscbr Alright. I've gone with your snapshot approach for those assertions. Let me know if you need anything else.

**Review by Chriscbr — APPROVED** [review]

File: src/awscdk/integration-test.ts [hunk 2]
```diff
@@ -84,14 +90,16 @@ export class IntegrationTest extends Component {
       '--no-version-reporting',
     ];
 
-    // add all feature flags
-    const features = [
-      ...FEATURE_FLAGS,
-      '@aws-cdk/core:newStyleStackSynthesis', // simplifies asset coordinates in synth output
-    ];
+    if (options.cdkDeps.cdkMajorVersion === 1) {
```
Comment (eladb) [inline src/awscdk/integration-test.ts:93]: I recon the same set of flags should be used in cdk.json of apps (not just integ tests), so maybe we can move this to AwsCdkDeps.

Something like `cdkDeps.featureFlags` property which will be major version aware.

**Pull request merged**